tests/unit/test_progress.py

```diff
@@ -1,17 +1,36 @@
 import logging
-from dvc.progress import Tqdm
+from dvc.progress import Tqdm, DVC_ISATTY
+import sys
+import mock
 
 
-def test_quiet(caplog, capsys):
+def test_quiet_logging(caplog, capsys):
     with caplog.at_level(logging.CRITICAL, logger="dvc"):
         for _ in Tqdm(range(10)):
             pass
         out_err = capsys.readouterr()
         assert out_err.out == ""
         assert out_err.err == ""
+
+
+def test_quiet_notty(caplog, capsys):
     with caplog.at_level(logging.INFO, logger="dvc"):
         for _ in Tqdm(range(10)):
             pass
         out_err = capsys.readouterr()
         assert out_err.out == ""
-        assert "0/10" in out_err.err
+        if DVC_ISATTY:
+            assert "0/10" in out_err.err
+        else:
+            assert out_err.err == ""
```

Comment:
This if doesn't make much sense if we only test one of its branches.

Reply:
may test both depending on env vars when the test is run

Reply:
But in `conftest.py` it's set to True for all tests.

Reply:
hmm yes but I feel this way is more future proof. I don't know what your test framework conventions are so can't really decide. At a minimum I think at least a comment would be required around these lines.